apps/dash-mapd-demo/app.py

```diff
@@ -0,0 +1,649 @@
+                    className="four columns",
+                    children=dcc.Loading(children=dcc.Graph(id="flights_hm")),
+                ),
+                html.Div(
+                    id="Flights-by-city-outer",
+                    className="four columns",
+                    children=dcc.Loading(children=dcc.Graph(id="value_by_city_graph")),
+                ),
+            ],
+        ),
+        html.Div(
+            id="bottom-row",
+            className="row",
+            children=[
+                html.Div(
+                    id="time-series-outer",
+                    className="four columns",
+                    children=dcc.Loading(
+                        children=dcc.Graph(
+                            id="flights_time_series",
+                            figure=generate_time_series_chart(
+                                "", "2018-01-01 00:00:00", "2018-01-08 00:00:00", "dep"
+                            ),
+                        )
+                    ),
+                ),
+                html.Div(
+                    id="Count_by_days_outer",
+                    className="four columns",
+                    children=dcc.Loading(children=dcc.Graph(id="count_by_day_graph")),
+                ),
+                html.Div(
+                    id="flight_info_table_outer",
+                    className="four columns",
+                    children=dcc.Loading(
+                        id="table-loading",
+                        children=dash_table.DataTable(
+                            id="flights-table",
+                            columns=[
+                                {"name": i, "id": i}
+                                for i in [
+                                    "flightnum",
+                                    "dep_timestamp",
+                                    "arr_timestamp",
+                                    "origin_city",
+                                    "dest_city",
+                                ]
+                            ],
+                            data=[],
+                            style_as_list_view=True,
+                            style_header={
+                                "textTransform": "Uppercase",
+                                "fontWeight": "bold",
+                            },
+                        ),
+                    ),
+                ),
+            ],
+        ),
+    ]
+)
+
+wk_map = {"Mon": 1, "Tues": 2, "Wed": 3, "Thu": 4, "Fri": 5, "Sat": 6, "Sun": 7}
+
+
+@app.callback(
+    output=Output("choropleth", "figure"),
+    inputs=[
+        Input("dropdown-select", "value"),
+        Input("date-picker-range", "start_date"),
+        Input("date-picker-range", "end_date"),
+    ],
+)
+def update_choro(dd_select, start, end):
+    # Update choropleth when dropdown or date-picker change
+    return generate_dest_choro(dd_select, start, end)
+
+
+def query_helper(state_query, dd_select, start, end, weekday_query):
```

Comment:
It might be good to move this function out of the "callbacks" section of the app. Maybe put it above?